src/core/net/ip6_address.hpp

```diff
@@ -96,6 +97,53 @@ class InterfaceIdentifier : public otIp6InterfaceIdentifier,
      */
     bool IsReserved(void) const;
 
+    /**
+     * This method generates and sets the Interface Identifier to a crypto-secure random byte sequence.
+     *
+     */
+    void GenerateRandom(void);
+
+    /**
+     * This method gets the Interface Identifier as a pointer to a byte array.
+     *
+     * @returns A pointer to a byte array (of size `kSize`) containing the Interface Identifier.
+     *
+     */
+    const uint8_t *GetAsByteArray(void) const { return mFields.m8; }
+
+    /**
+     * This method sets the Interface Identifier from a given byte array.
+     *
+     * @param[in] aBuffer    Pointer to an array containing the Interface Identifier. `kSize` bytes from the buffer
+     *                       are copied to form the Interface Identifier.
+     *
+     */
+    void SetFromByteArray(const uint8_t *aBuffer);
```

Comment:
Just curious - when do we make the method name explicit vs. use method overloading?

Reply:
Here are my thoughts/suggestions on this:

- For `set`ting an object in cases where a conversion/parsing is needed, I would say it'd better to require and use an explicit method name:
   - `Iid::SetFromExtAddress()`  is a good example (where we toggle the "local" bit to convert `ExtAddress` to IID),
   -  or `Dataset::SetFrom(const otOperationalDataset &)` is another good example. 
   - [in light of this, I would suggest renaming `Dataset::Set(const Message &, ...);` to `ReadFromMessage()`?]
- For cases where the value is directly used (no change/conversion), either way (use explicit method name or overload) is fine.
   - I personally prefer explicit name in this case as well (feel it is more readable and avoid any incorrect use), but we can leave it to developer.

Thoughts? @jwhui ?

Reply:
I generally prefer being explicit as well.